devdoc/channel_internal.md

```diff
@@ -63,17 +63,15 @@ MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_push, THANDLE(CHANNEL_INTER
 
 `channel_internal_close` schedules all pending operations to be abandoned.
 
-**SRS_CHANNEL_INTERNAL_43_094: [** `channel_internal_dispose` shall call `srw_lock_acquire_exclusive`. **]**
+**SRS_CHANNEL_INTERNAL_43_094: [** `channel_internal_close` shall call `srw_lock_acquire_exclusive`. **]**
 
-**SRS_CHANNEL_INTERNAL_43_095: [** `channel_internal_dispose` shall iterate over the list of pending operations and do the following: **]**
+**SRS_CHANNEL_INTERNAL_43_095: [** `channel_internal_close` shall iterate over the list of pending operations and do the following: **]**
 
  - **SRS_CHANNEL_INTERNAL_43_096: [** set the `result` of the `operation` to `CHANNEL_CALLBACK_RESULT_ABANDONED`. **]**
 
  - **SRS_CHANNEL_INTERNAL_43_097: [** call `threadpool_schedule_work` with `execute_callbacks` as `work_function`. **]**
 
-**SRS_CHANNEL_INTERNAL_43_100: [** `channel_internal_dispose` shall call `srw_lock_release_exclusive`. **]**
-
-**SRS_CHANNEL_INTERNAL_43_092: [** `channel_internal_dispose` shall release the reference to `THANDLE(CHANNEL_INTERNAL)`. **]**
+**SRS_CHANNEL_INTERNAL_43_100: [** `channel_internal_close` shall call `srw_lock_release_exclusive`. **]**
```

Comment:
>srw_lock_release_exclusive [](http://example.com/codeflow?start=72&length=26)

43_097+43_100 means that the threadpool miiiight execute its callbacks while the lock on the operations is taken, isn't it?

Now, what happens if the user (in one of its callbacks decides to call `channel_internal_pull` for example?

Reply:
The "user" of `channel_internal` is just `channel`. `channel` calls `channel_internal_close` when the user of `channel` releases their last reference to `THANDLE(CHANNEL)`. Since the user of `channel` has released their last reference, they can no longer call `channel_pull`, so `channel_internal_pull` can no longer be called either.